api/lib/infrastructure/repositories/certification-challenge-live-alert-repository.js

```diff
@@ -1,18 +1,52 @@
 import { DomainTransaction } from '../DomainTransaction.js';
 import { knex } from '../../../db/knex-database-connection.js';
+import { CertificationChallengeLiveAlert } from '../../domain/models/index.js';
+import { CertificationChallengeLiveAlertStatus } from '../../domain/models/CertificationChallengeLiveAlert.js';
+
+const certificationChallengeLiveAlertsTable = 'certification-challenge-live-alerts';
 
 const save = async function ({
   certificationChallengeLiveAlert,
   domainTransaction = DomainTransaction.emptyTransaction(),
 }) {
   const knexConn = domainTransaction.knexTransaction || knex;
-  return knexConn('certification-challenge-live-alerts').insert(certificationChallengeLiveAlert);
+  return knexConn(certificationChallengeLiveAlertsTable)
+    .insert({ ..._toDTO(certificationChallengeLiveAlert), updatedAt: new Date() })
+    .onConflict(['id'])
+    .merge();
 };
 
 const getByAssessmentId = async (assessmentId) => {
-  return knex('certification-challenge-live-alerts').where({
+  const certificationChallengeLiveAlertsDto = await knex(certificationChallengeLiveAlertsTable).where({
     assessmentId,
   });
+
+  return certificationChallengeLiveAlertsDto.map(_toDomain);
+};
+
+const getOngoingBySessionIdAndUserId = async ({ sessionId, userId }) => {
+  const certificationChallengeLiveAlertDtos = await knex('certification-courses')
+    .leftJoin('assessments', 'certification-courses.id', 'assessments.certificationCourseId')
+    .leftJoin(
+      certificationChallengeLiveAlertsTable,
+      `${certificationChallengeLiveAlertsTable}.assessmentId`,
+      'assessments.id',
+    )
+    .where({
+      'certification-courses.userId': userId,
+      'certification-courses.sessionId': sessionId,
+      [`${certificationChallengeLiveAlertsTable}.status`]: CertificationChallengeLiveAlertStatus.ONGOING,
+    })
+    .first();
+
+  return certificationChallengeLiveAlertDtos ? _toDomain(certificationChallengeLiveAlertDtos) : null;
```

Comment:
Suite du commentaire sur la gestion de l'erreur dans le repo et non dans le usecase .

Pour moi un `get` devrait trouver un résultat sinon tomber en erreur, un `find` lui peut retourner un tableau vide. 

du coup :
```suggestion
if(!certificationChallengeLiveAlertDtos) {
throw Error();
}
  return _toDomain(certificationChallengeLiveAlertDtos);
```

Reply:
Le fait de ne pas trouver de ongoing live alert n'est pas quelque chose d'incompatible avec le métier, il est préférable d'envoyer null plutôt qu'une erreur